### ISSUE TITLE
Consider injecting 'settings' and 'fail' properties to Groovy scripts

### ISSUE THREAD

**Issue by kirmerzlikin** [groovy/GMavenPlus#137]
Right now `AbstractToolsMojo` (GMavenPlus) injects same properties as `RuntimeMojoSupport` ([GMaven](https://github.com/groovy/gmaven/blob/gmaven-2.x/groovy-maven-plugin/src/main/java/org/codehaus/gmaven/plugin/RuntimeMojoSupport.java)) does except for two: *settings* and *fail*.

I think there are few reasons to consider injecting these properties in GMavenPlus as well:
- it will be easier for users of GMaven to migrate their projects to GMavenPlus if they don't need to re-write their Groovy scripts
- it will improve IDE support (e.g. Intellij IDEA) for GMavenPlus, as IDE developers will be able re-use autocompletion list (see [MavenGroovyPomScriptMemberContributor](https://github.com/JetBrains/intellij-community/blob/master/plugins/maven/src/main/java/org/jetbrains/idea/maven/plugins/groovy/MavenGroovyPomScriptMemberContributor.java))

**Comment by kirmerzlikin** [comment]
I took the liberty of creating a pull request with necessary changes - #138

**Comment by keeganwitt** [comment]
I apologize for the late reply.  I missed this somehow.  I'm taking a look now.

**Comment by keeganwitt** [comment]
Looking back, I think the reason I didn't include `settings` was that it was available from `session.settings` and leaving it out would tie up one less property that could potentially conflict with custom properties a user might wish to set.  This is also one reason we might want to reconsider adding it now: it could potentially break existing scripts.

And I think I didn't include `fail`, as I didn't understand the utility, given you can accomplish the same thing with a `System.exit` or something like `throw new MojoExecutionException()`.

Could you elaborate on how this could help the maintainability of IDEs like IntelliJ?  I didn't see another version of the class you mentioned.

**Comment by kirmerzlikin** [comment]
@keeganwitt 

I agree with you about the `settings`, it seems redundant when we have it in `session.settings`.
Regarding `fail` - IMHO it just looks nicer.

As for IntelliJ maintainability:
It all started when I noticed that Intellij didn't have same set of features for GMavenPlus plugin as it had for GMaven. The main problem for me was that it didn't recognize `<script></script>` content as Groovy, so it didn't provide syntax highlighting, autocompletion etc. 
After some investigation I found out that this could be fixed (in Intellij code) by injecting Groovy language for GMavenPlus plugin. But with this injection Intellij also starts using autocompletion list from [MavenGroovyPomScriptMemberContributor](https://github.com/JetBrains/intellij-community/blob/master/plugins/maven/src/main/java/org/jetbrains/idea/maven/plugins/groovy/MavenGroovyPomScriptMemberContributor.java). And what bothers me about it, is that this list contains entries that aren't provided by GMavenPlus, namely `settings`, `fail` and `pom`, which means that using IDE suggestions may lead to script execution errors.
So I see two options for addressing this issue:
* fix Intellij with simple solution (see my PR - https://github.com/JetBrains/intellij-community/pull/1169), re-using same autocompletion list **and** making sure that most of autocompletion options can be used in scripts executed by GMavenPlus
* fix Intellij with different solution, configuring separate autocompletion list specifically for GMavenPlus plugin.

Although I don't know yet what this second *'different'* solution may be, I presume that it will be more complex and less maintainable than first one.
